esp-hal/src/dma/gdma.rs

```diff
@@ -297,122 +243,177 @@ impl<const N: u8> RegisterAccess for Channel<N> {
-    }
-
-    fn unlisten_in_eof() {
-        Self::in_int()
-            .ena()
-            .modify(|_, w| w.in_suc_eof().clear_bit());
-    }
-
-    fn unlisten_out_eof() {
-        Self::out_int()
-            .ena()
-            .modify(|_, w| w.out_total_eof().clear_bit());
-    }
-
-    fn listen_ch_in_done() {
-        Self::in_int().ena().modify(|_, w| w.in_done().set_bit())
-    }
-
-    fn clear_ch_in_done() {
-        Self::in_int()
-            .clr()
-            .write(|w| w.in_done().clear_bit_by_one());
-    }
-
-    fn is_ch_in_done_set() -> bool {
-        Self::in_int().raw().read().in_done().bit()
+    fn listen_out(interrupts: impl Into<EnumSet<DmaTxInterrupt>>) {
+        Self::out_int().ena().modify(|_, w| {
+            for interrupt in interrupts.into() {
+                match interrupt {
+                    DmaTxInterrupt::TotalEof => w.out_total_eof().set_bit(),
+                    DmaTxInterrupt::DescriptorError => w.out_dscr_err().set_bit(),
+                    DmaTxInterrupt::Eof => w.out_eof().set_bit(),
+                    DmaTxInterrupt::Done => w.out_done().set_bit(),
+                };
+            }
+            w
+        })
+    }
+
+    fn unlisten_out(interrupts: impl Into<EnumSet<DmaTxInterrupt>>) {
+        Self::out_int().ena().modify(|_, w| {
+            for interrupt in interrupts.into() {
+                match interrupt {
+                    DmaTxInterrupt::TotalEof => w.out_total_eof().clear_bit(),
+                    DmaTxInterrupt::DescriptorError => w.out_dscr_err().clear_bit(),
+                    DmaTxInterrupt::Eof => w.out_eof().clear_bit(),
+                    DmaTxInterrupt::Done => w.out_done().clear_bit(),
+                };
+            }
+            w
+        })
     }
 
-    fn unlisten_ch_in_done() {
-        Self::in_int().ena().modify(|_, w| w.in_done().clear_bit());
-    }
+    fn is_listening_out() -> EnumSet<DmaTxInterrupt> {
+        let mut result = EnumSet::new();
 
-    fn is_listening_ch_in_done() -> bool {
-        Self::in_int().ena().read().in_done().bit()
-    }
+        let int_ena = Self::out_int().ena().read();
+        if int_ena.out_total_eof().bit_is_set() {
+            result |= DmaTxInterrupt::TotalEof;
+        }
+        if int_ena.out_dscr_err().bit_is_set() {
+            result |= DmaTxInterrupt::DescriptorError;
+        }
+        if int_ena.out_eof().bit_is_set() {
+            result |= DmaTxInterrupt::Eof;
+        }
+        if int_ena.out_done().bit_is_set() {
+            result |= DmaTxInterrupt::Done;
+        }
 
-    fn listen_in_descriptor_error() {
-        Self::in_int()
-            .ena()
-            .modify(|_, w| w.in_dscr_err().set_bit())
+        result
     }
 
-    fn unlisten_in_descriptor_error() {
-        Self::in_int()
-            .ena()
-            .modify(|_, w| w.in_dscr_err().clear_bit())
-    }
+    fn available_out_interrupts() -> EnumSet<DmaTxInterrupt> {
```

Comment:
the naming seems to be somewhat misleading - `available` sounds like "supported" but this really gets the currently set interrupt flags (same for available_in_interrupts)

Reply:
Fair, I'll just remove the `available` prefix then

Reply:
Sorry to bring this up after you've removed them, but `pending_in/out_interrupts` would probably be even better.